index.html

```diff
@@ -82,6 +128,32 @@ <h1>liczba wyborcow w danych okregach wyborczych</h1>
             }
         }
 
+        function displayClosestDistricts(distances, currentSquaredDistance) {
+            var closestLowerDistrict = null;
+            var closestHigherDistrict = null;
+
+            for (var i = 0; i < distances.length; i++) {
+                if (distances[i].distance < currentSquaredDistance) {
+                    closestLowerDistrict = distances[i];
+                } else if (distances[i].distance > currentSquaredDistance) {
+                    closestHigherDistrict = distances[i];
+                    break; 
+                }
+            }

```

Comment:
Good try! Unfortunately, this is not how you find 2 values that are closest to some other value.

Let's imagine the following:

```ts
const distances = [1, 9, 2, 8, 4, 6, 3, 7,];
const currentSquaredDistance = 5;
```

I'm skipping the `.distance` fields in `distances` so it's easier to see

The closest values are:

* lower: `4`
* higher : `6`

Your algorithm finds:

* lower: `1`
* higher: `9`

The problems:
1. It stops analyzing the rest of the array after it finds an element that is higher than the `currentSquaredDistance`. The closest elements could be later in the array, but the algorithm skips it.
  
    The `break` is unnecessary.

2. When it finds a new element that is lower/higher, it does not compare if the previously found `closestLowerDistrict` and `closestHigherDistrict` are already closer than the new element. The algorithm always overwrites the `closest...` values with the newly found value. Thus, even if you fixed the first problem, it would say that `3` and `7` are the closest, because those appear as last values that are lower/higher than `currentSquaredDistrict`.

    You need to take `closestLowerDistrict` into account in the `if` condition.
    You need to take `closestHigherDistrict` into account in the `else if` condition.

Reply:
Nevermind, I just understood that you are sorting the districts by distances earlier (in another function). This is not obvious, but I believe it could work fine. I suggest adding a comment why it needs to work this way.

By the way, IMO it would be better if you just found 2 districts that are closest by choosing the first 2 items in the array. Since you sorted them by distance, it means that the first 2 items in the array are closest (of course, after you fix the way the distance is calculated, which I mentioned in another comment).